### ISSUE TITLE
bug in rule 206 (Variables should have spaces before and after)

### ISSUE THREAD

**Issue by infothrill** [ansible/ansible-lint#498]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible --version
ansible 2.7.9
  config file = None
  configured module search path = [u'/home/infothrill/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/infothrill/src/git/ansible-role-hd_idle/.tox/ansible27/lib/python2.7/site-packages/ansible
  executable location = /home/infothrill/src/git/ansible-role-hd_idle/.tox/ansible27/bin/ansible
  python version = 2.7.16 (default, Mar  4 2019, 09:01:38) [GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.11.45.5)]

ansible-lint --version

ansible-lint 4.1.0
```

- ansible installation method: source (via tox)
- ansible-lint installation method: source (via tox)

# Desired Behaviour

Ansible-lint should not raise error `[206] Variables should have spaces before and after: {{ var_name }}`

# Actual Behaviour (Bug report only)

Running ansible-lint 4.1.0 on a minimal playbook containing only:
```
---
- name: "Ansible-lint bug in rule 206"
  get_url:
    url: "{{ url }}"
    dest: "{{ compile_dir }}/pkg-{{ version }}.tgz"
    checksum: "{{ checksum }}"
```
Results in:

```
[206] Variables should have spaces before and after: {{ var_name }}
bug.yml:5
    dest: "{{ compile_dir }}/pkg-{{ version }}.tgz"
```
This does not seem correct, as both variables have spaces around them.

**Comment by netzvieh** [comment]
Hi @infothrill, I can't reproduce this with ansible and ansible-lint installed via pip with the provided example task. Though of course you are correct in your expectation that 206 shouldn't match in this case. Maybe there's an other invisible character in there?

Could you check with `cat -A bug.yml`?

**Comment by infothrill** [comment]
@netzvieh thx for the hint with `cat -A`. I trusted vim (visually) a bit too much. There was a unicode "no-break space" (U+00A0) followed by a regular ascii space inside the curly braces:
```
cat -A bug.yml
---$
- name: "Ansible-lint bug in rule 206"$
  get_url:$
    url: "{{ url }}"$
    dest: "{{M-BM- compile_dir }}/pkg-{{ version }}.tgz"$
    checksum: "{{ checksum }}"$
```
That being said, it could be interesting to provide some kind of hint for such corner cases in the error message. I haven't looked at the implementation, so I don't know how hard that would be. Either way, this is not a bug in ansible-lint. Can be closed for my sake. thx again!

**Comment by awcrosby** [comment]
Thanks both for looking into this, closing this.
